assignments/prototypes.js

```diff
@@ -7,14 +9,33 @@
   
   Each constructor function has unique properites and methods that are defined in their block comments below:
 */
-  
+
+function dice(sides, numberOfDie) {
+    let working = [];
+    let min = 1;
+    for (let i = 0; i < numberOfDie; i++) {
+        working.push(Math.floor(Math.random() * (sides - min + 1)) + min);
+    }
+    return working.reduce((total, items) => total + items, 0);
+}
+
 /*
   === GameObject ===
   * createdAt
   * dimensions
   * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
 */
 
+function GameObject(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+    this.destroy = function(opponentName, damage, weapon) {
+        return `${
+            this.name
+        } was killed by ${opponentName}  after taking ${damage} damage from ${weapon}.`;
```

Comment:
You should use default values or have an if statement here since this isn't only necessarily used for characters who are attacked by other characters. For example:
```javascript
this.destroy = function(opponentName = 'World', damage = Infinity, weapon = 'World') {
  return `${
    this.name
  } was killed by ${opponentName} after taking ${damage} damage from ${weapon}.`;
};
```

Reply:
You can see where this would apply with this log on line 199:
```javascript
console.log(swordsman.destroy());
```